### ISSUE TITLE
`export { version } from "pkg/package.json"` results in invalid d.ts files

### ISSUE THREAD

**Issue by juliusmarminge** [huozhi/bunchee#467]
Repro: https://github.com/juliusmarminge/bunchee-export-pkgjson
```ts
export { version } from "../package.json";
```

generates the following d.ts files:
```ts
var version = "1.0.0";

export { version };
```

which is invalid:
![CleanShot 2024-02-28 at 13 48 58@2x](https://github.com/huozhi/bunchee/assets/51714798/6cfac0a7-c877-4c4f-8d61-9b633295a04f)

I think it should be
```ts
declare const version = "1.0.0";

export { version };
```
?

**Comment by huozhi** [comment]
Looks like because you put `"module": "index.ts",` in the package.json and it's being picked up as ESM output, which breaks the code. I rewrote the exports handling and added a test in #470 , it looks good. Let's verify it with incoming release 🙏

**Comment by huozhi** [comment]
Fixed in 5.0.0-beta.1
